test/model/src/include/model/driver/kv_workload_runner_wt.h

```diff
@@ -0,0 +1,345 @@
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef MODEL_DRIVER_KV_WORKLOAD_RUNNER_WT_H
+#define MODEL_DRIVER_KV_WORKLOAD_RUNNER_WT_H
+
+#include <memory>
+#include <shared_mutex>
+#include <unordered_map>
+#include "model/driver/kv_workload.h"
+#include "model/core.h"
+#include "wiredtiger.h"
+
+namespace model {
+
+/*
+ * kv_workload_runner_wt --
+ *     The workload context for WiredTiger.
+ */
+class kv_workload_runner_wt {
+
+protected:
+    /*
+     * session_context --
+     *     The WiredTiger session
+     */
+    class session_context {
+
+        using cursor_id_t = unsigned;
+        static constexpr cursor_id_t k_cursors_per_table = 16;
+
+    public:
+        /*
+         * session_context::session_context --
+         *     Create the
+         */
+        inline session_context(kv_workload_runner_wt &workload_context, WT_SESSION *session)
+            : _session(session), _workload_context(workload_context)
+        {
+        }
+
+        /*
+         * session_context::~session_context --
+         *     Destroy the context, alongside the corresponding resources.
+         */
+        ~session_context();
+
+        /*
+         * session_context::session --
+         *     Get the session.
+         */
+        inline WT_SESSION *
+        session() const noexcept
+        {
+            return _session;
+        }
+
+        /*
+         * session_context::cursor --
+         *     Get a cursor. Create one if it does not already exist. Use the second argument to get
+         *     and/or create additional cursors for the given table.
+         */
+        WT_CURSOR *cursor(table_id_t table_id, unsigned table_cur_id = 0);
+
+    private:
+        /*
+         * session_context::cursor_id --
+         *     Get a cursor ID.
+         */
+        static inline cursor_id_t
+        cursor_id(table_id_t table_id, unsigned table_cur_id)
+        {
+            if (table_cur_id < 0 || table_cur_id >= k_cursors_per_table)
+                throw model_exception("Cursor ID out of range");
+            return (cursor_id_t)(table_id * k_cursors_per_table + table_cur_id);
+        }
+
+    private:
+        WT_SESSION *_session;
+        kv_workload_runner_wt &_workload_context;
+
+        std::unordered_map<cursor_id_t, WT_CURSOR *> _cursors; /* One cursor per table. */
```

Comment:
The comment /* One cursor per table */ is no longer true.

Reply:
Nice catch.